src/components/dialogs/contingency-list/filter-based/contingency-list-filter-based-form.tsx

```diff
@@ -52,61 +43,76 @@ const equipmentTypes: string[] = [
-    }) as unknown as FilterSubEquipments[];
-
-    const handleFilterRowClick = useCallback((clickedFilterId: string, currentFilterId: string | null) => {
-        if (clickedFilterId !== currentFilterId) {
-            setSelectedFilterId(clickedFilterId);
-        }
-    }, []);
 
-    const filterEquipmentTypes =
-        selectedFilterId &&
-        filtersSubEquipments?.find((value) => value[ContingencyFieldConstants.FILTER_ID] === selectedFilterId)?.[
-            ContingencyFieldConstants.SUB_EQUIPMENT_TYPES
-        ];
+    const panelGroupRef = useRef<ImperativePanelGroupHandle>(null);
 
-    const handleEquipmentRowClick = useCallback(
-        (equipmentType: string, isEquipmentSelected: boolean, selectedFilterIdP: string | null) => {
-            const currentSubEquipmentsByFilters: FilterSubEquipments[] = getValues(
-                ContingencyFieldConstants.SUB_EQUIPMENT_TYPES_BY_FILTER
-            );
-            setValue(
-                ContingencyFieldConstants.SUB_EQUIPMENT_TYPES_BY_FILTER,
-                currentSubEquipmentsByFilters.map((value) => {
-                    if (value[ContingencyFieldConstants.FILTER_ID] === selectedFilterIdP) {
-                        // we either add or remove the clicked equipment from the list of subEquipments
-                        // depending on if the equipment is already selected or not
-                        const updatedSubEquipments = isEquipmentSelected
-                            ? value[ContingencyFieldConstants.SUB_EQUIPMENT_TYPES].filter(
-                                  (subEquipment: string) => subEquipment !== equipmentType
-                              )
-                            : [...value[ContingencyFieldConstants.SUB_EQUIPMENT_TYPES], equipmentType];
+    useEffect(() => {
+        const panelGroup = panelGroupRef.current;
+        if (substationAndVLFilters.length > 0 && !isSubOrVlFilterIncluded) {
+            setIsSubOrVlFilterIncluded(true);
+            if (panelGroup) {
+                panelGroup.setLayout([60, 40]);
+            }
+        } else if (substationAndVLFilters.length === 0 && isSubOrVlFilterIncluded) {
+            setIsSubOrVlFilterIncluded(false);
+            if (panelGroup) {
+                panelGroup.setLayout([33, 67]);
+            }
+        }
+    }, [substationAndVLFilters, setIsSubOrVlFilterIncluded, isSubOrVlFilterIncluded]);
 
-                        return {
-                            ...value,
-                            [ContingencyFieldConstants.SUB_EQUIPMENT_TYPES]: updatedSubEquipments,
-                        };
-                    }
-                    // for the other filters, we just return the value as is
-                    return value;
-                }),
-                { shouldDirty: true }
-            );
-            setIsDataOutdated(true);
-        },
-        [setValue, getValues]
-    );
+    // TODO : uncomment when useEffectEvent will be available
```

Comment:
I'm not a fan of such todo. Rather keep it in a branch or a PR. I guess there are plenty of hooks in our code that can be replaced with useEffectEvent()